src/core/thread/thread_netif.cpp

```diff
@@ -129,7 +129,18 @@ ThreadError ThreadNetif::GetLinkAddress(Ip6::LinkAddress &address) const
 
 ThreadError ThreadNetif::RouteLookup(const Ip6::Address &source, const Ip6::Address &destination, uint8_t *prefixMatch)
 {
-    return mNetworkDataLeader.RouteLookup(source, destination, prefixMatch, NULL);
+    ThreadError error;
+    uint16_t rloc;
+
+    SuccessOrExit(error = mNetworkDataLeader.RouteLookup(source, destination, prefixMatch, &rloc));
+
+    if (rloc == mMleRouter.GetRloc16())
+    {
+        error = kThreadError_NoRoute;
+    }
```

Comment:
Do I get this correct? 

We return an error if the  `rloc` of the dst is same as ours (this is I guess the case where the msg should be passed up to host, right?), and then we pass up everything that we have no route for to host (in the `else` part in the `ForwardMessage()`).